models/discordactions.js

```diff
@@ -114,32 +117,83 @@ const updateDiscordImageForVerification = async (userDiscordId) => {
   }
 };
 
-const getNumberOfMemberForGroups = async (groups = []) => {
+/**
+ * Enriches group data with membership information for a given Discord ID.
+ *
+ * @param {string} discordId - The Discord ID of the user.
+ * @param {Array<object>} groups - Array of group objects to process.
+ * @returns {Promise<Array<object>>} - An array of group objects with enriched information.
+ */
+const enrichGroupDataWithMembershipInfo = async (discordId, groups = []) => {
   try {
     if (!groups.length) {
       return [];
     }
+
+    const groupCreatorIds = groups.reduce((ids, group) => {
+      if (group.createdBy) {
+        ids.add(group.createdBy);
+      }
+      return ids;
+    }, new Set());
+
+    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
     const roleIds = groups.map((group) => group.roleid);
+    const groupsToUserMappings = await fetchGroupToUserMapping(roleIds);
+    const roleIdToCountMap = {};
 
-    const snapshots = await memberRoleModel.where("roleid", "in", roleIds).get();
-    const roleCount = {};
+    groupsToUserMappings.forEach((groupToUserMapping) => {
+      // Count how many times roleId comes up in the array.
+      // This says how many users we have for a given roleId
```

Comment:
These comments are important?

Reply:
Yes.
I added those as per this  https://github.com/Real-Dev-Squad/website-backend/pull/1413#discussion_r1297589755